pkg/run/trace_lifecycle.go

```diff
@@ -398,6 +393,129 @@ func (l traceLifecycle) OnFunctionCancelled(ctx context.Context, md sv2.Metadata
 	}
 }
 
+func (l traceLifecycle) OnFunctionSkipped(
+	ctx context.Context,
+	md sv2.Metadata,
+	s execution.SkipState,
+) {
+	ctx = l.extractTraceCtx(ctx, md, true)
+
+	start := time.Now()
+	if !md.Config.StartedAt.IsZero() {
+		start = md.Config.StartedAt
+	}
+
+	// spanID should always exists
+	spanID, err := md.Config.GetSpanID()
+	if err != nil {
+		// generate a new one here to be used for subsequent runs.
+		// this could happen for runs that started before this feature was introduced.
+		sid := NewSpanID(ctx)
+		spanID = &sid
+	}
+
+	runID := md.ID.RunID
+	slug := md.Config.FunctionSlug()
+
+	evtIDs := make([]string, len(md.Config.EventIDs))
+	for i, e := range md.Config.EventIDs {
+		evtIDs[i] = e.String()
+	}
+
+	// NOTE: generate the trigger span since it's skipped
+	{
+		_, trigger := NewSpan(ctx,
+			WithScope(consts.OtelScopeTrigger),
+			WithName(consts.OtelSpanTrigger),
+			WithTimestamp(ulid.Time(runID.Time())),
+			WithSpanAttributes(
+				attribute.String(consts.OtelSysAccountID, md.ID.Tenant.AccountID.String()),
+				attribute.String(consts.OtelSysWorkspaceID, md.ID.Tenant.EnvID.String()),
+				attribute.String(consts.OtelSysAppID, md.ID.Tenant.AppID.String()),
+				attribute.String(consts.OtelSysFunctionID, md.ID.FunctionID.String()),
+				attribute.String(consts.OtelSysFunctionSlug, md.Config.FunctionSlug()),
+				attribute.Int(consts.OtelSysFunctionVersion, md.Config.FunctionVersion),
+				attribute.String(consts.OtelAttrSDKRunID, runID.String()),
+				attribute.Int64(consts.OtelSysFunctionStatusCode, enums.RunStatusScheduled.ToCode()),
+				attribute.String(consts.OtelSysEventIDs, strings.Join(evtIDs, ",")),
+			),
+		)
+		defer trigger.End()
+
+		schedule := md.Config.CronSchedule()
+		if schedule != nil {
+			trigger.SetAttributes(attribute.String(consts.OtelSysCronExpr, *schedule))
+		}
+
+		batchID := md.Config.BatchID
+		if batchID != nil {
+			trigger.SetAttributes(
+				attribute.String(consts.OtelSysBatchID, batchID.String()),
+				attribute.Int64(consts.OtelSysBatchTS, int64(batchID.Time())),
+			)
+		}
+		if md.Config.DebounceFlag() {
+			trigger.SetAttributes(attribute.Bool(consts.OtelSysDebounceTimeout, true))
```

Comment:
Looks like we can now show this in the UI right?

Reply:
yes. the data has always been there. it's just never referenced on the UI before.